src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -332,13 +335,32 @@ public function denormalize($data, string $type, string $format = null, array $c
             }
 
             $value = $this->validateAndDenormalize($resolvedClass, $attribute, $value, $format, $context);
```

Comment:
`AbstractObjectNormalizer::validateAndDenormalize()` will either:
* return the value of `$value` is it's a valid builtin type
* return the result of `SerializerInterface::denormalize()` if `$value` is an object
* throw an exception of type `NotNormalizableValueException` if `$value` is an invalid builtin type

Validation errors are thrown as `NotNormalizableValueException`, so what do you think about:
```suggestion
        try {
            $value = $this->validateAndDenormalize($resolvedClass, $attribute, $value, $format, $context);
        } catch (NotNormalizableValueException $validationException) {
            if (!($context[self::COLLECT_INVARIANT_VIOLATIONS] ?? false)) {
                throw $validationException;
            }

            $violation = new InvariantViolation($value, $validationException->getMessage(), $validationException);
            $value = DenormalizationResult::failure([$attribute => [$violation]]);
        }
```

Reply:
Actually, if the `collect_invariant_violations` context option is enabled, nested denormalizers are all expected to always return an instance of `DenormalizationResult`. If an exception is thrown, it means that some nested denormalizer does not support the option or some unrecoverable failure happened, in both case this should make the whole denormalization process fail with an exception.

Maybe we should even throw an exception when the option is enabled and a nested denormalizer does not return an instance of `DenormalizationResult`?

Reply:
I got your point but this method does not only return the value of a denormalizer.
So if you don't get a `DenormalizationResult` result it might not be because of an "invalid" denormalizer.
If the attribute is of a builtin type but the value is invalid then a `NotNormalizableValueException` will be thrown.

All that is done internally in the `AbstractObjectNormalizer::validateAndDenormalize()` method, so I see only two options:
1. update the method manually: it's tedious and you'll have to check if the collect option is activated every single time
2. catch the exceptions as proposed, if I understood correctly those exceptions will only be thrown from this method since the one from other normalizer will be handled then and returned as `DenormalizationResult`.

Reply:
I chose option 1 to avoid catching `NotNormalizableValueException` from nested denormalizers.